src/System.Device.Gpio/System/Device/I2c/I2cDevice.cs

```diff
@@ -1,13 +1,21 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Globalization;
+
 namespace System.Device.I2c;
 
 /// <summary>
 /// The communications channel to a device on an I2C bus.
 /// </summary>
 public abstract partial class I2cDevice : IDisposable
 {
+    /// <summary>
+    /// This event is raised when <see cref="QueryComponentInformation"/> of the base class is called. This is can be used for bindings
+    /// to intercept that call on their own I2C device, so they can add additional information
+    /// </summary>
+    public event Action<ComponentInformation>? OnQueryComponentInformation;
```

Comment:
This should be just callback rather than event especially given 1:1 relationship with device. Why not `Func<ComponentInformation>` instead?

Reply:
ok my question in the end got answered by the code below. First comment is still valid though

Reply:
btw I like this approach much more than inheritance initially proposed but I'll suggest to do only printing in the initial PR and in the following one we can focus on the approach

Reply:
So you mean to remove this for now?